apps/src/javalab/JavabuilderConnection.js

```diff
@@ -66,7 +68,7 @@ export default class JavabuilderConnection {
       '/javabuilder/access_token',
       requestData,
       /* checkProjectEdited */ true
-    );
+    ).then(this.onConnectDone);
```

Comment:
Why not just add this directly to `connectJavabuilderHelper`?

Reply:
Yeah, so I did originally have this in `connectJavabuilderHelper` chained on to the ajax call (`$.ajax(...).finally(this.onConnectDone)`) but I realized we early return in that function, so rather than calling it in two places in the same function, i thought it would be slightly cleaner to convert this to an async function. That way hypothetically if we add another early return or some other completion condition that's not the ajax call, we don't need to also remember to add `this.onConnectDone`.

However yeah It does mean that we end up calling it in three places (each time `connectJavabuilderHelper` is called) so I was also considering refactoring a bit more and consolidating those functions but that felt like a bit of overkill for this change. But I'm happy to go down that route if it would be clearer?

Reply:
Got it -- I'm fine if you decide to leave as-is, but would it be much work just to wrap connectJavabuilderHelper in something that calls it and the Redux callback (and call that in each `connectJavabuilder...` function)? It feels like something we'd always want to call after connecting to Javabuilder, so I could see us forgetting to do that or something in the future.

Reply:
I agree, I think it would be best if we can consolidate this so it doesn't need to be called in multiple places.

Reply:
Sounds good, I'll put this in a wrapper function to avoid duplicating